### PR TITLE
Add openconfig model for ip prefix and nexthop counters

### PR DESCRIPTION
**Pull request by fzhuatca**
In the jointly shared meeting notes between google/arista,  20211215, new openconfig models for ip prefix and nexthop counters are proposed.

"[Mayukh] OC model for Prefix counters + Encap counters
For Prefix counters our proposal is to have a separate sub container within the state container per prefix. The counter sub container will have:
Packets-forwarded
octets-forwarded
If we need to disambiguate primary vs backup, then we will need to add two more attributes in here:
Packets-forwarded-backup
Octets-forwarded-backup
So overall our proposal would be:
Add a subcontainer called Counters under the State container which will contain the 4 attributes:
Packets-forwarded
Octets-forwarded
Packets-forwarded-backup
Octets-forwarded-backup

For In-in-ip encapsulation counters our proposal would be to add the counter attributes to the nexthop state container
Packets-forwarded
Octets-forwarded
So our proposal would be to add a sub containers called Counters which will have:
Packets-forwarded
Octets-forwarded"

### CODE REVIEW

**Comment by dplore** [comment]
Thanks @fzhuatca, LGTM.  @earies would you like to cross examine.

**Comment by dplore** [comment]
Reference pyang tree output:

```
module: openconfig-network-instance
  +--rw network-instances
     +--rw network-instance* [name]
        +--ro afts
        |  +--ro ipv4-unicast
        |  |  +--ro ipv4-entry* [prefix]
        |  |     +--ro prefix    -> ../state/prefix
        |  |     +--ro state
        |  |        +--ro prefix?                            oc-inet:ipv4-prefix
        |  |        +--ro counters
        |  |        |  +--ro packets-forwarded?          oc-yang:counter64
        |  |        |  +--ro octets-forwarded?           oc-yang:counter64
        |  |        |  +--ro packets-forwarded-backup?   oc-yang:counter64
        |  |        |  +--ro octets-forwarded-backup?    oc-yang:counter64
        |  |        +--ro next-hop-group?                    -> /network-instances/network-instance/afts/next-hop-groups/next-hop-group/state/id
        |  |        +--ro next-hop-group-network-instance?   -> /network-instances/network-instance/config/name
        |  |        +--ro entry-metadata?                    binary
        |  |        +--ro origin-protocol?                   identityref
        |  |        +--ro decapsulate-header?                oc-aftt:encapsulation-header-type
        |  +--ro ipv6-unicast
        |  |  +--ro ipv6-entry* [prefix]
        |  |     +--ro prefix    -> ../state/prefix
        |  |     +--ro state
        |  |        +--ro prefix?                            oc-inet:ipv6-prefix
        |  |        +--ro counters
        |  |        |  +--ro packets-forwarded?          oc-yang:counter64
        |  |        |  +--ro octets-forwarded?           oc-yang:counter64
        |  |        |  +--ro packets-forwarded-backup?   oc-yang:counter64
        |  |        |  +--ro octets-forwarded-backup?    oc-yang:counter64
        |  |        +--ro next-hop-group?                    -> /network-instances/network-instance/afts/next-hop-groups/next-hop-group/state/id
        |  |        +--ro next-hop-group-network-instance?   -> /network-instances/network-instance/config/name
        |  |        +--ro entry-metadata?                    binary
        |  |        +--ro origin-protocol?                   identityref
        |  |        +--ro decapsulate-header?                oc-aftt:encapsulation-header-type
```

**Comment by dplore** [comment]
@fzhuatca Adding a new container 'counters' makes this a breaking change.  Rather than adding this container, can we add the two leafs in the existing grouping?  It is preferred to avoid breaking changes unless there is a strong motivation.

A non-breaking change could look like:
```
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/octets-forwarded
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/octets-forwarded-backup
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/packets-forwarded
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/packets-forwarded-backup
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/decapsulate-header
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/entry-metadata
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/next-hop-group
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/next-hop-group-network-instance
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/origin-protocol
/openconfig-network-instance:network-instances/network-instance/afts/ipv4-unicast/ipv4-entry/state/prefix
```

**Review by dplore — CHANGES_REQUESTED** [review]
The new container 'counters' makes this a breaking change. Rather than adding this container, can we add the two leafs in the existing grouping? It is preferred to avoid breaking changes unless there is a strong motivation.

**Comment by dplore** [comment]
After discussion, there is precedent across many other models, for example interfaces to use a counters container.  For convenience subscribe can easily include or exclude the counters container which carries high frequency updates, as opposed to the prefix state which changes much less frequently.    So I am ok with the introduction of a new container for ip nexthop and ip prefix counters.

**Review by dplore — APPROVED** [review]

**Review by dplore — CHANGES_REQUESTED** [review]
SInce this is introducing a breaking change, the version should be incremented by a major version.  (ie: 1.0.0 instead of 0.10.0)

**Comment by dplore** [comment]
Hi @fzhuatca please also rebase/merge to master to fix the related pyangbind CI check which is failing.

**Comment by dplore** [comment]
Thank you for the contribution.  I have added this to the May 3, 2022 openconfig operators agenda.  Barring any objection, I expect to merge that day.

**Comment by rolandphung** [comment]
It looks like the version check that failed is due to https://github.com/openconfig/public/commit/3a5269e4e52d5ca42cd4240111cf224e23e8e381 (#439)

**Comment by fzhuatca** [comment]
Are we still on track to merge this PR's? Thanks! @dplore 
(BTW, It looks like the version check that failed is due to https://github.com/openconfig/public/pull/439 . If it needs to be fixed in this PR, please let me know the details. Thanks)

File: release/models/aft/openconfig-aft-common.yang [hunk 1]
```diff
@@ -136,6 +142,14 @@ submodule openconfig-aft-common {
 
           uses aft-common-entry-nexthop-state;
           uses aft-labeled-entry-state;
+
+          container counters {
+            config false;
```
[the remark this answers is not in the document]
Comment (fzhuatca) [reply release/models/aft/openconfig-aft-common.yang]: Thanks for reviewing the PR. Have updated accordingly.

File: release/models/aft/openconfig-aft-ipv4.yang [hunk 2]
```diff
@@ -132,7 +138,14 @@ submodule openconfig-aft-ipv4 {
           "The IPv4 destination prefix that should be matched to
           utilise the AFT entry.";
     }
-    uses aft-common-entry-state;
+    uses aft-common-entry-state {
+      augment counters {
+        description
+          "counter for the backup NHG.";
```
[the remark this answers is not in the document]
Comment (fzhuatca) [reply release/models/aft/openconfig-aft-ipv4.yang]: Thanks for reviewing the PR. Have updated accordingly.

**Comment by dplore** [comment]
It would help me and others if you would submit a fix for #439 ,  thank you for offering!

**Comment by fzhuatca** [comment]
Ok. Copied the version number, 0.8.0, and the corresponding revision information from openconfig-rib-bgp-tables.yang to the rest of openconfig-rib-bgp-*attribute.yang. It could pass the test/check now.

**Review by dplore — APPROVED** [review]
LGTM

**Pull request merged**